pytest/tests/sanity/sync_ban.py

```diff
@@ -19,97 +19,112 @@
+                "secs": 3,
+                "nanos": 0
+            },
+            "header_sync_stall_ban_timeout": {
+                "secs": 5,
+                "nanos": 0
+            }
         },
+        "tracked_shards": [0]
     }
-}
-node1_config = {
-    "consensus": {
-        "header_sync_initial_timeout": {
-            "secs": 3,
-            "nanos": 0
+
+    should_sync = Value('i', False)
+    nodes = start_cluster(
+        1, 1, 1, None, [["epoch_length", EPOCH_LENGTH]], {
+            0: node0_config,
+            1: node1_config
         },
-        "header_sync_stall_ban_timeout": {
-            "secs": 5,
-            "nanos": 0
-        }
-    },
-    "tracked_shards": [0]
-}
-nodes = start_cluster(1, 1, 1, None, [["epoch_length", EPOCH_LENGTH]], {
-    0: node0_config,
-    1: node1_config
-}, Handler)
-
-utils.wait_for_blocks(nodes[0], target=110, poll_interval=2)
-
-should_sync.value = True
-
-logger.info("sync node 1")
-
-start = time.time()
-
-tracker0 = utils.LogTracker(nodes[0])
-tracker1 = utils.LogTracker(nodes[1])
-
-while True:
-    assert time.time() - start < TIMEOUT
-
-    if should_ban:
-        if tracker1.check(BAN_STRING):
-            break
-    else:
-        cur_height = nodes[0].get_latest_block().height
-        node1_height = nodes[1].get_latest_block().height
-        if (abs(node1_height - cur_height) < 5 and
-                status1['sync_info']['syncing'] is False):
-            break
-    time.sleep(2)
-
-if not should_ban and (tracker0.check(BAN_STRING) or
-                       tracker1.check(BAN_STRING)):
-    assert False, "unexpected ban of peers"
-
-logger.info('shutting down')
-time.sleep(10)
+        functools.partial(Handler,
+                          should_sync=should_sync,
+                          should_ban=should_ban))
+
+    utils.wait_for_blocks(nodes[0], target=30, poll_interval=2)
+
+    should_sync.value = True
+
+    logger.info("sync node 1")
+
+    start = time.time()
+
+    tracker0 = utils.LogTracker(nodes[0])
+    tracker1 = utils.LogTracker(nodes[1])
+
+    while True:
+        assert time.time() - start < TIMEOUT
+
+        if should_ban:
+            if tracker1.check(BAN_STRING):
+                break
+        else:
+            cur_height = nodes[0].get_latest_block().height
+            node1_height = nodes[1].get_latest_block().height
+            status1 = nodes[1].get_status()
+            print(
+                f"Sync: node 1 at block {node1_height}, node 0 at block {cur_height}; waiting for node 1 to catch up"
+            )
+            if (abs(node1_height - cur_height) < 5 and
+                    status1['sync_info']['syncing'] is False):
+                break
+        time.sleep(2)
+
+    if not should_ban and (tracker0.check(BAN_STRING) or
+                           tracker1.check(BAN_STRING)):
+        assert False, "unexpected ban of peers"
+
+    # logger.info('shutting down')
+    # time.sleep(10)
```

Comment:
What about this?

Reply:
It isn't needed. I should've removed the two lines though, I'll do that next time :)

Note that this test was not even enabled before this PR.